examples/pytorch/infograph/semisupervised.py

```diff
@@ -0,0 +1,222 @@
+    )
+
+    # Step 4: training epochs =============================================================== #
+    sup_loss_all = 0
+    unsup_loss_all = 0
+    consis_loss_all = 0
+
+    best_val_error = 99999
+    best_test_error = 99999
+    for epoch in range(args.epochs):
+        ''' Training '''
+        model.train()
+        lr = scheduler.optimizer.param_groups[0]['lr']
+
+        iteration = 0
+        sup_loss_all = 0
+        unsup_loss_all = 0
+        consis_loss_all = 0
+
+        for sup_data, unsup_data in zip(train_loader, unsup_loader):
+            sup_graph, sup_target = sup_data
+            unsup_graph, _ = unsup_data
+
+            sup_graph = sup_graph.to(args.device)
+            unsup_graph = unsup_graph.to(args.device)
+            
+            sup_nfeat, sup_efeat = sup_graph.ndata['attr'], sup_graph.ndata['edge_attr']
+            unsup_nfeat, unsup_efeat, unsup_graph_id = unsup_graph.ndata['attr'],\
+                                                       unsup_graph.edata['edge_attr'], unsup_graph.edata['graph_id']
+
+            sup_target = (sup_target - mean) / std
+            sup_target = sup_target.to(args.device)
+
+            optimizer.zero_grad()
+
+            sup_loss = F.mse_loss(model(sup_graph, sup_nfeat, sup_efeat), sup_target)
+            unsup_loss, consis_loss = model.unsup_forward(unsup_graph, unsup_nfeat, unsup_efeat, unsup_graph_id)
+
+            loss = sup_loss + unsup_loss + args.reg * consis_loss
+
+            loss.backward()
+
+            sup_loss_all += sup_loss.item()
+            unsup_loss_all += unsup_loss.item()
+            consis_loss_all += consis_loss.item()
+
+            optimizer.step()
+
+        print('Epoch: {}, Sup_Loss: {:4f}, Unsup_loss: {:.4f}, Consis_loss: {:.4f}' \
+            .format(epoch, sup_loss_all, unsup_loss_all, consis_loss_all))
+
+        model.eval()
+
+        val_error = 0
+        test_error = 0
+        
+
+        for val_graphs, val_targets in val_loader:
+
+            val_graph = val_graphs.to(args.device)
+            val_nfeat, val_efeat = val_graph.ndata['attr'], val_graph.edata['edge_attr']
+            val_target = (val_targets - mean) / std
+            val_target = val_target.to(args.device)
+
+            val_error += (model(val_graph, val_nfeat, val_efeat) * std - val_target * std).abs().sum().item()
+
+        val_error = val_error / val_num
+        scheduler.step(val_error)
+
+        if val_error < best_val_error:
+            best_val_error = val_error
+
+            for test_graphs, test_targets in test_loader:
+
+                test_graph = test_graphs.to(args.device)
+                test_nfeat, test_efeat = test_graph.ndata['attr'], test_graph.edata['edge_attr']
+
+                test_target = (test_targets - mean) / std
+                test_target = test_target.to(args.device)
+
+                test_error += (model(test_graph, test_nfeat, test_efeat) * std - test_target * std).abs().sum().item()
```

Comment:
Same for the test set. You can actually write a function and use that for the evaluation on both the validation set and the test set.